roles/keycloak_clustered_deployment/tasks/main.yml

```diff
@@ -1,25 +1,24 @@
-# Big shoutout to my homies at DigitalOcean from which I could base the initial docker setup on
-# (https://www.digitalocean.com/community/tutorials/how-to-use-ansible-to-install-and-set-up-docker-on-ubuntu-18-04 but we're debian gang instead)
-
-- name: Keycloak clustered deployment
-  hosts: docker_swarm_manager
-
 - name: Create Postgres database storage directory
   file:
     path: '{{ postgres_data_directory }}'
     state: directory
     mode: '0740'
 
 - name: Create keycloak container stack
-  docker_stack:
-    name: '{{ keycloak_container_name }}'
-    state: present
+  include_role:
+    name: 'docker_stack_create'
+  vars:
+    stack_name: '{{ keycloak_stack_name }}'
+    service_imbude:
+      traefik:
+        keycloak:
+          auth:
+            enabled: TRUE
+            route_rule: '{{ keycloak_traefik_rule }}'
+            service_port: 8080
+            sticky_loadbalance: TRUE
     compose:
-    - version: "3"
-      networks:
-        traefik:
-          external:
-            name: '{{ traefik_docker_network }}'
+      version: "3.2"
```

Comment:
that reminds me, I should figure out a better compose version to use, since everything has to use the same version as docker_stack_create (because docker is lame). 3.2 was chosen because it both had a feature we needed, and because 3.3 made a breaking syntax change to a feature I was using.